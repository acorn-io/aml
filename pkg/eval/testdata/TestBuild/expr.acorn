bin: 1 + 1
uni: !3
parens: (!false)
ref: foo
sel: foo.bar
index: "string"[0]
slice: "str"[0:1]
slice: "str"[:1]
slice: "str"[1:]
slice: "str"[:]