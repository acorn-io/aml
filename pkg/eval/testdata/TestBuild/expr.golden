&eval.File{Body: &eval.Struct{
	Position: eval.Position{
		Filename: "expr.acorn",
		Line:     1,
		Column:   1,
	},
	Fields: []eval.Field{
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{
					Value: value.String("bin"),
				},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Line:     1,
					Column:   1,
				},
			},
			Value: &eval.Op{
				Operator: value.Operator("+"),
				Left:     eval.Value{Value: value.Number("1")},
				Right:    eval.Value{Value: value.Number("1")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   7,
					Line:     1,
					Column:   8,
				},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Line:     1,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("uni")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   11,
					Line:     2,
					Column:   1,
				},
			},
			Value: &eval.Op{
				Unary:    true,
				Operator: value.Operator("!"),
				Left:     eval.Value{Value: value.Number("3")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   16,
					Line:     2,
					Column:   6,
				},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   11,
				Line:     2,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("parens")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   19,
					Line:     3,
					Column:   1,
				},
			},
			Value: &eval.Parens{Expr: &eval.Op{
				Unary:    true,
				Operator: value.Operator("!"),
				Left:     eval.Value{Value: value.Boolean(false)},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   28,
					Line:     3,
					Column:   10,
				},
			}},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   19,
				Line:     3,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("ref")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   36,
					Line:     4,
					Column:   1,
				},
			},
			Value: &eval.Lookup{
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   41,
					Line:     4,
					Column:   6,
				},
				Key: "foo",
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   36,
				Line:     4,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("sel")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   45,
					Line:     5,
					Column:   1,
				},
			},
			Value: &eval.Selector{
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   50,
					Line:     5,
					Column:   6,
				},
				Base: &eval.Lookup{
					Pos: eval.Position{
						Filename: "expr.acorn",
						Offset:   50,
						Line:     5,
						Column:   6,
					},
					Key: "foo",
				},
				Key: eval.Value{Value: value.String("bar")},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   45,
				Line:     5,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("index")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   58,
					Line:     6,
					Column:   1,
				},
			},
			Value: &eval.Index{
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   65,
					Line:     6,
					Column:   8,
				},
				Base:  eval.Value{Value: value.String("string")},
				Index: eval.Value{Value: value.Number("0")},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   58,
				Line:     6,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("slice")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   77,
					Line:     7,
					Column:   1,
				},
			},
			Value: &eval.Slice{
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   89,
					Line:     7,
					Column:   13,
				},
				Base:  eval.Value{Value: value.String("str")},
				Start: eval.Value{Value: value.Number("0")},
				End:   eval.Value{Value: value.Number("1")},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   77,
				Line:     7,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("slice")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   95,
					Line:     8,
					Column:   1,
				},
			},
			Value: &eval.Slice{
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   107,
					Line:     8,
					Column:   13,
				},
				Base: eval.Value{Value: value.String("str")},
				End:  eval.Value{Value: value.Number("1")},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   95,
				Line:     8,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("slice")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   112,
					Line:     9,
					Column:   1,
				},
			},
			Value: &eval.Slice{
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   124,
					Line:     9,
					Column:   13,
				},
				Base:  eval.Value{Value: value.String("str")},
				Start: eval.Value{Value: value.Number("1")},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   112,
				Line:     9,
				Column:   1,
			},
		},
		&eval.KeyValue{
			Key: eval.FieldKey{
				Key: eval.Value{Value: value.String("slice")},
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   129,
					Line:     10,
					Column:   1,
				},
			},
			Value: &eval.Slice{
				Pos: eval.Position{
					Filename: "expr.acorn",
					Offset:   141,
					Line:     10,
					Column:   13,
				},
				Base: eval.Value{Value: value.String("str")},
			},
			Pos: eval.Position{
				Filename: "expr.acorn",
				Offset:   129,
				Line:     10,
				Column:   1,
			},
		},
	},
}}
